### ISSUE TITLE
Detail view of a svg image is not loading in asset browser

### ISSUE THREAD

**Issue by enwin** [statamic/v2-hub#1740]
### Expected behaviour
Detail view should display the svg file with all the informations (Dimensions, Size, Last Modified) and the ability to add a Title and an Alt Text.

### Actual behaviour
Detail view display the loader indefinitely. With a 500 error code on the ajax call to `/cp/assets/-path to the svg file-`

### Steps to reproduce
1.  Go to the asset page in the control panel
2.  Upload a svg file (you can try [this one](https://cdn.switch-company.com/statamic-svg-examples/logo-lastfm.svg))
3.  Once uploaded, double click the image to display the details
4. Watch the loader spinning indefinitely

### Server Details

**Operating System**:
macOS Sierra 10.12.6

**PHP Version:**
PHP 7.0.23 (cli)

**Statamic Version:**
2.8.2

### Logs
The 500 error on the ajax call to `cp/assets/-path to the svg file-`:
<details>
ErrorException in DimensionBuilder.php line 122:
array_splice() expects parameter 1 to be array, boolean given
</details>

### Quick fix
In the DimensionBuilder.php file the `getImageDimensions` method only use `getimagesize` to get the image' size. Which is fine but doesn't work on svgs.

Here's my take on the fix:

```php
private function getImageDimensions($path)
    {
        if( Path::extension($path) === 'svg' ){
          $svgfile = simplexml_load_file($path);
          $viewBox = explode(" ", $svgfile['viewBox']);

          // width and height are the last to entries in the viewBox tag
          return array_splice($viewBox, -2);
        }

        $size = getimagesize($path);

        // getimagesize may return more than two keys. We only want the first two.
        return array_splice($size, 0, 2);
    }
```

I'm not a php person so feel free to adapt my snippet 😄

**Comment by jasonvarga** [comment]
I have a feeling this is will come down to a PHP version thing. Uploading the svg you provided half works. It doesn't display a preview, but there are no errors. It just doesn't consider an svg an image.

We'll need to dig a little deeper!

**Comment by enwin** [comment]
I did some other tests and I realised that I had set the svg files to be considered as images (testing #1730). I changed the `isImage()` method in `statamic/core/Assets/Asset.php` to include the svg extension and didn't rolled back the file before filling this bug.

With svg considered as images the detail view tries to get the dimension and a preview compared as the default behaviour where it doesn't. That's the reason the bug appeared.

I'm really sorry about wasting your time. 😞

Closing this issue.
I guess this code could be used if the svg gets implemented as an image later on 😅
